components/scream/src/physics/p3/p3_constants.hpp

```diff
@@ -105,6 +105,9 @@ constexpr Scalar Constants<Scalar>::Tmelt;
 template <typename Scalar>
 constexpr Scalar Constants<Scalar>::Tol;
 
+template <typename Scalar>
+constexpr Scalar Constants<Scalar>::Pi;
+  
```

Comment:
I was surprised that I needed to add the lines just above this comment. There are lots of constants defined in this file and only a few of them have this additional constexpr definition, but I only seem to be able to access constants with this definition. Is nobody using the other constants, or am I not accessing the consts data correctly in impl.hpp? And for ease of reference, in impl.hpp I'm using C::Pi to access.

Reply:
None of these should be in this header as I believe the pattern in general breaks [ODR](https://en.cppreference.com/w/cpp/language/definition). (Here I think we're saved from breaking ODR by the fact that the values are templated.) Best practice IMO is simply to write
```
const auto Pi = C::Pi;
```
at the top of a function definition that uses the constant Pi.